doc/ipythonconsole.rst

```diff
@@ -2,91 +2,145 @@
-   * If using a version of Spyder before 3.3.0, ``ipython kernel`` should work to launch the kernel, albeit without certain Spyder-specific features.
+Spyder also supports several types of specialized consoles.
+A `Cython console`_ allows you to use the Cython language to speed up your code and call C functions directly from Python.
+A `Sympy console`_ enables creating and displaying symbolic math expressions right inside Spyder.
+Finally, a `Pylab console`_ loads common Numpy and Matplotlib functions by default; while this is deprecated and not recommended for new code, it can be used for older scripts that need it.
+
+.. _Cython console: https://cython.org/#documentation
+.. _Sympy console: https://docs.sympy.org/latest/index.html
+.. _Pylab console: https://matplotlib.org/faq/usage_faq.html#matplotlib-pyplot-and-pylab-how-are-they-related
+
+.. image:: images/console/console-special.gif
+   :alt: Spyder showing opening a new special Console
+
+
+
+============
+Options menu
+============
+
+The options menu allows you to inspect your current environment variables and the contents of your system's ``PATH`` in Spyder :doc:`variableexplorer` browsers with :guilabel:`Show environment variables` and :guilabel:`Show sys.path contents` respectively, and you can have each kernel display how long it has been running with :guilabel:`Show elapsed time`.
+
+.. image:: images/console/console-options-menu.png
+   :alt: Spyder IPython Console with options menu
+
+You can also change the name of the current :guilabel:`IPython console` tab by simply double-clicking it, or with the :guilabel:`Rename tab` option.
+
+.. image:: images/console/console-rename.gif
+   :alt: Spyder IPython Console showing renaming console
+
+
+
+.. _connecting-external-kernel:
+
+==================================
+Using external kernels with Spyder
+==================================
+
+You can connect to external local and remote kernels (including those managed by QtConsole sessions or the Jupyter Notebook) through the :guilabel:`Connect to an existing kernel` dialog under the :guilabel:`Consoles` menu, the pane menu or its context menu.
+External kernels still support all of the :ref:`features<console-features>` of Spyder's IPython console.
+To work with Spyder, a compatible version of the ``spyder-kernels`` package :ref:`must be installed <starting-kernel-problems-ref>` in the environment or machine in which the external kernel is running.
+
+.. image:: images/console/console-menu.png
+   :alt: Connect to external kernel dialog of the Spyder IPython console
 
-#. Copy the connection file (:file:`{jupyter/runtime/dir/path}/kernel-{pid}.json`) to the machine you're running Spyder on (if remote) or note its location (if local).
+
+Connect to a local kernel
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To connect to a local kernel that is already running (e.g. under Jupyter notebook),
```

Comment:
```suggestion
To connect to a local kernel that is already running (e.g. one started by the Jupyter notebook),
```

Be more specific for clarity.

Reply:
Agreed. Tiny wording tweak as above (remove "the"), 

```suggestion
To connect to a local kernel that is already running (e.g. one started by Jupyter notebook),
```